int32 maxCount
int32 count